File: 3870-count-commas-in-range/solution.py
class Solution:
    def countCommas(self, n: int) -> int:
        total_commas = 0
        threshold = 1000

        # Add the contribution for each threshold of 1000^k
        while n >= threshold:
            total_commas += n - threshold + 1
            threshold *= 1000

        return total_commas

File: 3870-count-commas-in-range/solution.cpp
class Solution {
public:
    int countCommas(int n) {
        long long totalCommas = 0;
        long long threshold = 1000;

        // Add the contribution for each threshold of 1000^k
        while (n >= threshold) {
            totalCommas += (n - threshold + 1);
            threshold *= 1000;
        }

        return totalCommas;
    }
};